### DISCUSSION TITLE
How to make XRDP to only listen on eth0 interface and not on Wi-Fi

### DISCUSSION THREAD

**Discussion by tapir** [neutrinolabs/xrdp#2445] [Q&A] [answered]
I have 2 computers (one windows another linux) that are connected to the same Wi-Fi router and through Wi-Fi I can successfully connect from the windows client to my linux XRDP machine.

However, I'd like to use Wi-Fi only for internet access and do the RDP communication via the ethernet connection hence I've connected the computers alsa via a dumb switch (with static IPs and a separate subnet compared to wifi)

Question is how can I make XRDP to use eth0 instead of wlan?

**Comment by matt335672** [comment, marked as answer]
This may depend on your xrdp version.

On recent ones, you need to change the `port=` setting in `/etc/xrdp/xrdp.ini`. By default xrdp listens on port 3389 on all interfaces, and you can see this with `ss` :-

```
ss -alnp \( sport = :3389 \)
Netid  State   Recv-Q  Send-Q   Local Address:Port   Peer Address:Port Process  
tcp    LISTEN  0       2                    *:3389              *:* 
```

On a virtual machine with enp1s0 having an IP address  of aa.bb.cc.dd, I can add this to `/etc/xrdp/xrdp.ini`:-

```
port=tcp://aa.bb.cc.dd:3389
```

Then:-

```
$ ss -alnp \( sport = :3389 \)
Netid  State   Recv-Q  Send-Q   Local Address:Port   Peer Address:Port Process  
tcp    LISTEN  0       2        aa.bb.cc.dd:3389        0.0.0.0:*             
```

You can also achieve the same result with the firewall on your machine. This is in many ways a better administrative solution as you get to control exactly what can get in on each interface without having to modify many different config files for different apps.

**Comment by tapir** [comment]
thanks a lot!
